### PR TITLE
Updates to Talljoe's Keymaps

### PR DESCRIPTION
**Pull request by Talljoe**
## Description

Long overdue merge of keymaps and their mapping to new keyboards.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Talljoe** [comment]
I don't know why the checks are failing. It seems like an error with the script?

File: keyboards/atreus/keymaps/talljoe/config.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/atreus/keymaps/talljoe/config.h:19]: This doesn't actually exist, and should be removed. 
```suggestion
```

File: keyboards/jd45/keymaps/talljoe/config.h [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
 
 #include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/jd45/keymaps/talljoe/config.h:18]: This doesn't actually exist, and should be removed. 
```suggestion
```

File: keyboards/atreus/keymaps/talljoe/config.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/atreus/keymaps/talljoe/config.h:21]: This no longer exists, as this is the default behavior now. 
```suggestion
```

File: keyboards/primekb/prime_e/keymaps/talljoe/config.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/primekb/prime_e/keymaps/talljoe/config.h:19]:
```suggestion
```

File: keyboards/thevankeyboards/bananasplit/keymaps/talljoe/config.h [hunk 5]
```diff
@@ -1,5 +1,20 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
 
 #include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/thevankeyboards/bananasplit/keymaps/talljoe/config.h:19]:
```suggestion
```

File: keyboards/thevankeyboards/minivan/keymaps/talljoe-minivan/config.h [hunk 6]
```diff
@@ -1,5 +1,20 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
 
 #include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/thevankeyboards/minivan/keymaps/talljoe-minivan/config.h:19]:
```suggestion
```

File: keyboards/tmo50/keymaps/talljoe/keymap.c [hunk 7]
```diff
@@ -0,0 +1,44 @@
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/tmo50/keymaps/talljoe/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/tmo50/keymaps/talljoe/keymap.c [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+#define LED_MASK (_BV(PB0) | _BV(PB1) | _BV(PB2) | _BV(PB3))
+#define WHITE (_BV(PB0))
+#define YELLOW (_BV(PB1))
+#define MAGENTA (_BV(PB2))
+#define RED (_BV(PB3))
+
+void matrix_scan_keymap(void) {
+  uint32_t lights = WHITE;
+
+  switch(biton32(layer_state))
```
Comment (drashna) [inline keyboards/tmo50/keymaps/talljoe/keymap.c:17]: This has been aliased now, as we support 8 and 16 bit layer state bitmasks.
```suggestion
  switch(get_highest_layer(layer_state))
```

File: keyboards/tmo50/keymaps/talljoe/keymap.c [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      lights |= YELLOW;
+      break;
+    case _NUM:
+      lights |= MAGENTA;
+      break;
+    case _ADJUST:
+      lights |= RED;
+      break;
+    default: {
+      uint32_t default_layer = biton32(default_layer_state);
```
Comment (drashna) [inline keyboards/tmo50/keymaps/talljoe/keymap.c:29]: Pretty sure you don't need more than an 8 bit int here :)
```suggestion
      uint8_t default_layer = get_highest_layer(default_layer_state);
```

File: keyboards/tmo50/keymaps/talljoe/keymap.c [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      lights = 0; // no white LED by default.
+      if(default_layer & 1)
```
Comment (drashna) [inline keyboards/tmo50/keymaps/talljoe/keymap.c:31]: Alternatively, you can do `layer_state_cmp(default_layer_state, 1)` for this, and the others.

File: keyboards/tmo50/tmo50.c [hunk 11]
```diff
@@ -51,34 +51,51 @@ void led_set_kb(uint8_t usb_led) {
 	led_set_user(usb_led);
 }
 
-uint32_t layer_state_set_user(uint32_t state)
+uint32_t layer_state_set_kb(uint32_t state)
```
Comment (drashna) [inline keyboards/tmo50/tmo50.c:54]: Would prefer that these keyboard changes were in a separate PR. 

Also, 
```suggestion
layer_state_t layer_state_set_kb(layer_state_t state)
```

Comment (Talljoe) [reply keyboards/tmo50/tmo50.c]: Done. #10150

File: keyboards/tmo50/keymaps/talljoe/keymap.c (hunk 10 above)
Comment (Talljoe) [reply keyboards/tmo50/keymaps/talljoe/keymap.c:31]: Here I'm actually displaying the binary equivalent of the default layer on the LEDs. Using `layer_state_cmp` for that would probably be confusing so I'll leave as is.

**Comment by Talljoe** [comment]
Thank you for the feedback! I've pulled the TMO changes out to a separate pull request (#10150) and based this PR off of that branch.

File: keyboards/tmo50/keymaps/talljoe/keymap.c (hunk 10 above)
Comment (drashna) [reply keyboards/tmo50/keymaps/talljoe/keymap.c:31]: Okay, not a problem, at all.

**Comment by Talljoe** [comment]
Hi!  I've resolved all outstanding feedback. Can I get a re-review, please? Thank you!

**Comment by drashna** [comment]
Missed the `QMK_KEYBOARD_CONFIG_H` comment (probably because it's hidden by default ...)

**Comment by Talljoe** [comment]
> probably because it's hidden by default

Well that's useful! :)

Fixed.

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**